File: token_comp.py

class Token:
    TK_IDENTIFIER = 0
    TK_NUMBER = 1
    TK_OPERATOR = 2
    TK_PUNCTUATION = 3
    TK_ASSIGN = 4

    TK_TEXT = [
        "IDENTIFICADOR", "NÚMERO", "OPERADOR", "PUNTUACIÓN", "ASIGNACIÓN"
    ]

    def __init__(self, type=None, text=""):
        self.type = type
        self.text = text
        self.line = None
        self.column = None

    def set_type(self, type):
        self.type = type

    def get_type(self):
        return self.type

    def set_text(self, text):
        self.text = text

    def get_text(self):
        return self.text

    def set_line(self, line):
        self.line = line

    def get_line(self):
        return self.line

    def set_column(self, column):
        self.column = column

    def get_column(self):
        return self.column
    
    def to_string(self):
        type_text = Token.TK_TEXT[self.type] if self.type is not None else "None"
        return print(f"Token [type={type_text}, text={self.text}, line={self.line}, column={self.column}]")

File: token_comp.cpp
#include "token_comp.h"
#include <iostream>

using namespace std;

const int Token::Tk_identifier =0;
const int Token::Tk_number =1;
const int Token::Tk_operator = 2;
const int Token::Tk_punctuantion = 3;
const int Token::Tk_assign = 4;
const int Token::Tk_hola = 5;

const string Token::Tk_text[]= {"IDENTIFICADOR", "NUMERO","OPERADOR","PUNTUACION","ASIGNACION"};

Token::Token(int type, const std::string& text, int line, int column)
    : type(type), text(text), line(line), column(column) {}

void Token::set_type(int type) {
    this->type = type;
}

int Token::get_type() const {
    return type;
}

void Token::set_text(const string& text) {
    this->text = text;
}

const string& Token::get_text() const {
    return text;
}

void Token::set_line(int line) {
    this->line = line;
}

int Token::get_line() const {
    return line;
}

void Token::set_column(int column) {
    this->column = column;
}

int Token::get_column() const {
    return column;
}

void Token::to_string() const {
    string type_text = (type != -1 && type < 5) ? Tk_text[type] : "None";
    cout << "Token [type=" << type_text << ", text=" << text << ", line=" << line << ", column=" << column << "]" << endl;
}